client/fakes/softlayer_client_fake.go

```diff
@@ -191,7 +209,7 @@ func (fslc *FakeSoftLayerClient) DoRawHttpRequestWithObjectMask(path string, mas
 		return []byte{}, fslc.DoRawHttpRequestError
 	}
 
-	if fslc.DoRawHttpRequestResponse != nil && len(fslc.DoRawHttpRequestResponses) == 0 {
+	if fslc.DoRawHttpRequestResponse != nil {
```

Comment:
Can you elaborate why you removed the check for `len(fslc.DoRawHttpRequestResponses) == 0`? Also, have you ran the integration tests with this?